clients/tabby-chat-panel/src/index.ts

```diff
@@ -84,8 +210,20 @@ export interface ClientApiMethods {
 
   onKeyboardEvent: (type: 'keydown' | 'keyup' | 'keypress', event: KeyboardEventInit) => void
 
-  // find symbol definition location by hint filepaths and keyword
-  onLookupSymbol?: (hintFilepaths: string[], keyword: string) => Promise<SymbolInfo | undefined>
+  /**
+   * Find the target symbol and return the symbol information.
+   * @param symbol The symbol to find.
+   * @param hints The optional {@link LookupSymbolHint} list to help find the symbol. The hints should be sorted by priority.
+   * @returns The symbol information if found, otherwise undefined.
+   */
+  lookupSymbol?: (symbol: string, hints?: LookupSymbolHint[] | undefined) => Promise<SymbolInfo | undefined>
+
+  /**
+   * Open the target file location in the editor.
+   * @param target The target file location to open.
+   * @returns Whether the file location is opened successfully.
+   */
+  openInEditor: (target: FileLocation) => Promise<boolean>
```

Comment:
Should this be a method implemented by all clients? Or should it only be implemented by those who have this capability? It seems we already have a 'base' method called 'navigate', and 'base' implies it should be handled by all client IDEs.

Reply:
Yes, this method is intended to replace the existing `navigate` method.

I will add a new method `openExternal(url: string): Promise<boolean>` next, and replace all use of `navigate` with either `openInEditor` or `openExternal`, then deprecate the `navigate` method.